sklearn/manifold/t_sne.py

```diff
@@ -111,17 +160,147 @@ def _kl_divergence(params, P, alpha, n_samples, n_components):
     return kl_divergence, grad
 
 
-def _gradient_descent(objective, p0, it, n_iter, n_iter_without_progress=30,
+def _kl_divergence_error(params, P, neighbors, alpha, n_samples, n_components):
+    """t-SNE objective function: the absolute error of the
+    KL divergence of p_ijs and q_ijs.
+
+    Parameters
+    ----------
+    params : array, shape (n_params,)
+        Unraveled embedding.
+
+    P : array, shape (n_samples * (n_samples-1) / 2,)
+        Condensed joint probability matrix.
+
+    neighbors : array (n_samples, K)
+        The neighbors is not actually required to calculate the
+        divergence, but is here to match the signature of the
+        gradient function
+
+    alpha : float
+        Degrees of freedom of the Student's-t distribution.
```

Comment:
> I think it should be named degrees_of_freedom to make the code more readable

+1

> but as this is private API this is no big deal.

Still, it would make the code more readable.